lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2476,6 +2476,7 @@ IRGenDebugInfoImpl::emitFunction(const SILDebugScope *DS, llvm::Function *Fn,
   llvm::DITypeArray Error = nullptr;
   if (FnTy && (Opts.DebugInfoLevel > IRGenDebugInfoLevel::LineTables))
     if (auto ErrorInfo = FnTy->getOptionalErrorResult()) {
+      GenericContextScope scope(IGM, FnTy->getInvocationGenericSignature());
```

Comment:
And this is only needed when creating the function signature and not in the body of the function?

Reply:
We should verify that; it didn't show up as a crash when putting typed throws into the standard library, but it might be because we aren't providing correct debug info for SIL-level indirect error results.